client.go

```diff
@@ -200,6 +201,42 @@ func (c *Client) GetPNCounter(ctx context.Context, name string) (*PNCounter, err
 	return c.proxyManager.getPNCounter(ctx, name)
 }
 
+// GetDistributedObjectsInfo returns the information of all objects created cluster-wide.
+func (c *Client) GetDistributedObjectsInfo(ctx context.Context) ([]types.DistributedObjectInfo, error) {
+	if atomic.LoadInt32(&c.state) != ready {
+		return nil, hzerrors.ErrClientNotActive
+	}
+
+	request := codec.EncodeClientGetDistributedObjectsRequest()
+	resp, err := c.proxyManager.invokeOnRandomTarget(ctx, request, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	objects := codec.DecodeClientGetDistributedObjectsResponse(resp)
+	result := make([]types.DistributedObjectInfo, 0, len(objects))
+
+	cached := make(map[types.DistributedObjectInfo]struct{})
+	for _, o := range c.proxyManager.getCachedObjectsInfo() {
+		cached[o] = struct{}{}
+	}
```

Comment:
I think we shouldn't do anything with the proxy cache anymore.

Reply:
Isn't it still a good way of removing stale proxies?

Reply:
I think purging stale proxies is a separate concern. As a user, I wouldn't expect `GetObjectsInfo` call to mutate any internal state.

Reply:
Not sure about what to call the purge function though. If there's no immediate need from the Turbine team, maybe we can postpone adding it to another PR (unless you have a good name for that function) .

Reply:
Alright, fair enough.  Addressed in cd2887a

Reply:
Apart from the context of this pr, could it be handled via DistributedObjectListeners in proxy manager - i.e. purge the cache on object destroy or something?

Reply:
That can be an option, I can't think of a problem off the top of my head.